liblangutil/SemVerHandler.cpp

```diff
@@ -265,14 +271,16 @@ unsigned SemVerMatchExpressionParser::parseVersionPart()
 		{
 			c = currentChar();
 			if (v * 10 < v || v * 10 + static_cast<unsigned>(c - '0') < v * 10)
-				BOOST_THROW_EXCEPTION(SemVerError());
+				BOOST_THROW_EXCEPTION(SemVerError() << util::errinfo_comment("Invalid token in version pragma: "s + (c ? c : '\0')));
```

Comment:
Not sure if the `NULL` check for `c` is necessary here, and if yes, what example `pragma ...` statement would cause this to hit the null check branch, and if there is one, it would be nice to have a test for it.

Reply:
Yes. I agree that it is not needed. The value returned from `currentChar` and assigned to `c` is checked in the loop condition (line `270`) and should never return `0`.

Reply:
Then, I think the second part of the condition of the `if`, `v * 10 + static_cast<unsigned>(c - '0') < v * 10` should always be false.

Reply:
Also thinking that the whole condition is unnecessary... 
 - `v` is initialized in line `268` with a value in the range `[1, 9]`, guaranteed by the condition of the `if` in line `266`. So `v * 10 < v` should be false the first time the loop runs.
 - the loop condition always checks that `currentChar` is part of the same token and that it is in the range `[0,9]`, which when added to `v` should never make its value decrease. Then, `v * 10 + static_cast<unsigned>(c - '0') < v * 10` should always be false in every loop iteration. 
 - from the previous,  ``v * 10 < v` shall be false in loop iterations after the first.

Reply:
If in doubt, `solAssert` :)

Reply:
Anyway, I think the possible removal of the `if` block should be done in a separate PR. For now, I will just remove the ternary operator checking `NULL` for `c`.

Reply:
> Then, I think the second part of the condition of the `if`, `v * 10 + static_cast<unsigned>(c - '0') < v * 10` should always be false.

Not really. The `while` only ensures that `c` is a character between `'0'` and `'9'`. It does not ensure that adding the character to the number does not cause overflow in `v`.

I think that you can hit this case with this example:
```solidity
pragma solidity 4294967296;
```

`4294967296` is `2**32`. When you're at the final `6`, `v` is `429496729`. Multiplying it by `10` gives you `4294967290`, which still fits in `unsigned` and does not overflow so the first part of the `if` is false. But then adding `6` to that brings it over the edge so the second part of the `if` is true.

Reply:
Right, at the time I didn't think about extreme values and overflow.